api/types/accesslist/accesslist.go

```diff
@@ -173,6 +167,25 @@ func (a *AccessList) CheckAndSetDefaults() error {
 		return trace.BadParameter("grants must specify at least one role or trait")
 	}
 
+	// Deduplicate owners. The backend will currently prevent this, but it's possible that access lists
+	// were created with duplicated owners before the backend checked for duplicate owners. In order to
+	// ensure that these access lists are backwards compatible, we'll deduplicate them here.
+	ownerMap := map[string]bool{}
+	deduplicatedOwners := []Owner{}
+	for _, owner := range a.Spec.Owners {
+		if owner.Name == "" {
+			return trace.BadParameter("owner name is missing")
+		}
+
+		if ownerMap[owner.Name] {
+			continue
+		}
```

Comment:
💅 
We could initialize the map with a specific size:
`make(map[string]bool, len(a.spec.Owners))`

Another thing we could do is to use `struct{}` instead of `bool`
`make(map[string]struct{}, len(a.spec.Owners))`
And then check for existence instead of its value
```go
		if _, ok := ownerMap[owner.Name]; ok {
			continue
		}
```
The internet says it's faster 😅 
Also, I find existence checks more in line with what we need, and there's no doubt of what the map's value means.
Eg, what would a `false` value mean?
This is a really short-lived map, so I think that's ok eitherway 👍

Reply:
I've always found the map[string]bool to be more readable, but I can convert this into a map[string]struct{}!